$MirOS$

	This script is a pile of crap that should be replaced
	by a very short mksh script!

--- pc-bios/optionrom/signrom.sh.orig	Wed Sep 23 19:00:49 2009
+++ pc-bios/optionrom/signrom.sh	Sat Oct 17 20:31:50 2009
@@ -26,12 +26,14 @@ test "$1" -a "$2" || exit 1
 sum=0
 
 # find out the file size
-x=`dd if="$1" bs=1 count=1 skip=2 2>/dev/null | od -t u1 -A n`
+#x=`dd if="$1" bs=1 count=1 skip=2 2>/dev/null | od -t u1 -A n`
+x=$(dd if="$1" bs=1 count=1 skip=2 2>/dev/null | hexdump -ve '1/1 "%u"')
 #size=`expr $x \* 512 - 1`
-size=$(( $x * 512 - 1 ))
+size=$(( x * 512 - 1 ))
 
 # now get the checksum
-nums=`od -A n -t u1 -v "$1"`
+#nums=`od -A n -t u1 -v "$1"`
+nums=$(hexdump -ve '1/1 "%u "' "$1")
 for i in ${nums}; do
     # add each byte's value to sum
     sum=`expr $sum + $i`
